src/azure-cli/azure/cli/command_modules/role/_help.py

```diff
@@ -22,7 +22,14 @@
 helps['ad app create'] = """
 type: command
 short-summary: Create an application.
-long-summary: For more detailed documentation, see https://learn.microsoft.com/graph/api/resources/application
+long-summary: >-
+    IMPORTANT: The "az ad app create" command can modify an existing application or service principal if
+    another object shares the same display name. Display names aren't unique and can change, which
+    could result in credential loss or incorrect RBAC assignments. Use a unique object ID or app ID instead.
```

Comment:
It is only possible for the user to choose a unique **display name**. The user cannot choose to use a unique **object ID** or **app ID**. The **object ID** or **app ID** is generated by the MS Graph service.